### PR TITLE
feat: adds support for self-funded `useBlockProfiles`

### PR DESCRIPTION
**Pull request by reecejohnson**

### CODE REVIEW

File: packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts [hunk 1]
```diff
@@ -66,11 +71,10 @@ export class BlockProfilesGateway
       id: result.id,
       request,
       typedData: omitTypename(result.typedData),
-      fallback: this.createRequestFallback(request, result),
     });
   }
 
-  private async relayWithProfileManager(
+  private async broadcast(
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts:77]: Intentional?

File: examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx [hunk 2]
```diff
@@ -11,6 +11,15 @@ import { ErrorMessage } from '../components/error/ErrorMessage';
...
 import { ProfileCard } from './components/ProfileCard';
 
+function SponsoredFlag() {
+  return (
+    <label>
+      <input type="checkbox" name="sponsored" value="on" defaultChecked={true} />
+      sponsored
+    </label>
+  );
+}
```
Comment (cesarenaldi) [inline examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx:21]: I don't see the point in the level of indirection introduces by this component.

File: examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx [hunk 3]
```diff
@@ -40,21 +49,25 @@ function BlockOrUnblockProfileCard({ profile }: { profile: Profile }) {
...
+            const formData = new FormData(event.currentTarget);
             if (profile.operations.isBlockedByMe.isFinalisedOnchain === false) {
               alert(
                 'You cannot block this profile until the pending unblock operation is finalised onchain',
               );
               return;
             }
-            return block({ profiles: [profile] });
+            return block({ profiles: [profile], sponsored: formData.get('sponsored') === 'on' });
           }}
```
Comment (cesarenaldi) [inline examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx:63]: Why all this code is inlined here? Can we hoist it on a render-scope callback?

File: pnpm-lock.yaml
Comment (cesarenaldi) [inline pnpm-lock.yaml:1]: I don't see any obvious change to `package.json` files? What caused this change?

File: packages/react/src/transactions/useBlockProfiles.ts
Comment (cesarenaldi) [inline packages/react/src/transactions/useBlockProfiles.ts:1]: Can we add relevant TSDoc too?

File: .changeset/eleven-numbers-wave.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@lens-protocol/domain": minor
+"@lens-protocol/react": minor
+---
+
+adds support for self-funded `useBlockProfiles`
```
Comment (cesarenaldi) [inline .changeset/eleven-numbers-wave.md:6]:
```suggestion
**feat:** adds support for self-funded `useBlockProfiles`
```

for consistency.

File: packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts [hunk 5]
```diff
@@ -84,14 +88,19 @@ export class BlockProfilesGateway
 
     if (data.result.__typename === 'LensProfileManagerRelayError') {
       const result = await this.createTypedData(request);
-      const fallback = this.createRequestFallback(request, result);
+      const fallback = this.createSetBlockStatusCallDetails(result);
 
       return handleRelayError(data.result, fallback);
```
[the remark this answers is not in the document]
Comment (reecejohnson) [reply packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts:93]: Great, thanks

File: packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts (hunk 1 above)
Comment (reecejohnson) [reply packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts:77]: intentional, but only to keep consistency between gateways - i see they've been renamed to this now.

will revert back

File: examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx (hunk 2 above)
Comment (reecejohnson) [reply examples/web/src/profiles/UseBlockAndUnblockProfiles.tsx]: fair, removed

File: packages/react/src/transactions/adapters/profiles/__tests__/UpdateProfileManagerGateway.spec.ts
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/profiles/__tests__/UpdateProfileManagerGateway.spec.ts:1]: This test file content does not seem to match then name. Intentional or accidentally included?

**Pull request merged**